src/containers/WalletCreation/NameWalletForm.tsx

```diff
@@ -1,36 +1,44 @@
 import * as React from 'react'
 import { connect } from 'react-redux'
+import { Link } from 'react-router-dom'
 
 import { startWalletCreation, updateFormField } from '@actions'
 import { InputField } from '@components/InputField'
-import { RootState } from '@store'
-import { Link } from 'react-router-dom'
+import { Dispatch, RootState } from '@store'
+import { WALLET_CREATE_FORM_NAME } from '@types'
 
 const isValidWalletName = (name: string) => name.length > 0 && name.length <= 50
 const isValidPassword = (password: string) => password.length >= 12 && password.length <= 30
 
+interface OwnProps {
+  submitButtonText: string
+  onSubmitButtonClick(ev: React.MouseEvent): void
+} 
+
 function validCreateWalletState(state: RootState): boolean {
   const { name, password, confirmPassword } = state.createWallet.createForm
   return isValidWalletName(name) && isValidPassword(password) && password === confirmPassword
 }
-const mapStateToProps = (state: RootState) => ({
+const mapStateToProps = (state: RootState, { onSubmitButtonClick, submitButtonText }: OwnProps) => ({
   ...state.createWallet.createForm,
   canSubmit: validCreateWalletState(state),
+  onSubmitButtonClick,
+  submitButtonText,
 })
 
-const mapDispatchToProps = {
+const mapDispatchToProps = (dispatch: Dispatch) => ({
   updateFormField: (formField: keyof RootState['createWallet']['createForm'], fieldValue: string) =>
-    updateFormField({ fieldValue, formField, formName: 'WALLET_CREATE' }),
-  startWalletCreation,
-}
+    dispatch(updateFormField({ fieldValue, formField, formName: WALLET_CREATE_FORM_NAME })),
+  onSubmit: (ev: React.FormEvent) => {
+    ev.preventDefault()
+    dispatch(startWalletCreation())
+  },
```

Comment:
You still wouldn't need to wrap this in the dispatch as long as startWalletCreation was the thing being returned.

Reply:
ok cool